### PR TITLE
Include Core Data crash fix in 18.7

### PR DESCRIPTION
**Pull request by twstokes**
This adds the changes from https://github.com/wordpress-mobile/WordPress-iOS/pull/17488 (which has been merged to `develop`) to the 18.7 release branch.

To test:

See #17488

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * [***] Site Comments: Updated comment details with a fresh new look and capability to display rich contents. [#17466]
 * [**] Block editor: Image block: Add ability to quickly link images to Media Files and Attachment Pages [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3971]
 * [**] Block editor: Fixed a crash that could occur when copying lists from Microsoft Word. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/4174]
+* [***] Fixed crash where uploading image when offline crashes iOS app. [#17488]
```
Comment (AliSoftware) [inline RELEASE-NOTES.txt]: Quite the redundant wording: "a crash which crashes" 😅 

I'll let @mokagio decide but imho we shouldn't _yet again_ update the AppStore, Editorial'd release notes for this new entry at that point, especially so far after the code-freeze and as the release notes items from this list have already been thru Editorial and even translated in some locales already.

This is even more true this time around, because we've _already_ updated the Editorial'd release notes for 18.7 _twice_ sooner this week, which invalidated the translations that were already done each time… and was a bit annoying for the translators to see their translations invalidated multiple times in a row right after having submitted them in GlotPress (see the WPorg slack).

So TL;DR: even if we can keep this new item in `RELEASE-NOTES.txt` (for developers), I don't think we should run the lane to update the `.po` file (for end users).

**Review by AliSoftware — APPROVED** [review]

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (twstokes) [reply RELEASE-NOTES.txt]: > Quite the redundant wording: "a crash which crashes"

Agreed, that should be tweaked.

Comment (twstokes) [reply RELEASE-NOTES.txt]: I've tweaked and pushed a change @AliSoftware.

Comment (AliSoftware) [reply RELEASE-NOTES.txt]: Thanks!

[This warning is still true though](https://github.com/wordpress-mobile/WordPress-iOS/pull/17499#issuecomment-972971717). But it's still nice to have developer-facing notes with a good language even if they'll not be user-facing in the Editorial'd ones. 👍

Comment (twstokes) [reply RELEASE-NOTES.txt]: Makes sense @AliSoftware - thanks.

**Comment by mokagio** [comment]
We can discard the Buildkite failures.

I'm going to merge this.

**Comment by mokagio** [comment]
@twstokes   this has been bundled as part of [18.7 beta 1 (18.7.0.1)](https://github.com/wordpress-mobile/WordPress-iOS/pull/17562).

Thanks for your work 🙌

**Comment by twstokes** [comment]
Thank you @mokagio!
